source/concepts/recording.rst

```diff
@@ -25,6 +25,81 @@ Each stream is stored in a separate file.
 By convention, recordings of the frame stream have the ``.traj`` file extension,
 while recordings of the shared state stream  have the ``.state`` file extension.
 
+Recording format
+----------------
+
+The current version of the file format is version 2.
+Each recording file contains a header and a sequence of records.
+
+The header contains two fields, stored as little endian 8 bytes unsigned integers:
+
+* **a magic number, its value is 6661355757386708963**. This value was chosen arbitrarily and needs to be the first
+  8 bytes of the file to indicate it is indeed a NanoVer recording. A file without this magic number is not a NanoVer
+  recording, however one should keep in mind that a file that starts with that value could still not be a valid
+  recording and should handle errors accordingly.
+* **the version of the file format**. This version number dictates how the rest of the file will be written or parsed.
+  Any change to the file format needs to increment this file format version. The current version is 2.
+
+A record contains:
+
+* a timestamp encoded as a little endian 16 bytes unsigned integer that indicates the time, in microseconds,
+  since the beginning of the recording.
+  This timestamp indicates the timing of the records and allows synchronisation of a trajectory and a state recording.
+* the size, in bytes, of the record; encoded as an 8 bytes little endian unsigned integer.
+* the record itself as a protobuf message.
+
+In the case of a trajectory recording, each record contains a ``GetFrameResponse`` message.
+This message contains two fields: the frame index and the frame itself.
+The frame index is generally an integer that gets incremented each time the server register a frame to broadcast.
+However, its value is only significant when it is 0 as it means the frame needs to be reset;
+for instance because the server loaded a new simulation. The frame is a :ref:`FrameData <traj-and-frames>`.
```

Comment:
```suggestion
for instance because the server loaded a new simulation. The frame itself is defined by the :`FrameData <traj-and-frames>` class.
```

Reply:
Done in commit [52dc633](https://github.com/IRL2/nanover-docs/pull/113/commits/52dc633153e8d43308b50314cf888181b9394a23).